other_enrichment_methods/nbconverted/3_viz_pathway_ranking.py

```diff
@@ -0,0 +1,223 @@
+    )
+    
+    fig = pn.ggplot(plot_df, pn.aes(x='Test statistic', y='Percentile rank'))
+    fig += pn.geom_point()
+    fig += pn.geom_point(plot_df[plot_df['Percentile rank']>0.9],
+                         pn.aes(x='Test statistic', y='Percentile rank'),
+                         color='red'
+                        )
+    fig += pn.geom_text(pn.aes(label=
+                               [x if plot_df.loc[x,'Percentile rank']>0.9 else "" for x in plot_df.index]),
+                        ha='left',
+                        va='top',
+                        size=5
+                       )
+    fig += pn.labs(x = x_label,
+                y = 'Percentile of ranking',
+                title = f'{method_name} pathway statistics vs ranking')
+    fig += pn.theme_bw()
+    fig += pn.theme(
+        legend_title_align = "center",
+        plot_background=pn.element_rect(fill='white'),
+        legend_key=pn.element_rect(fill='white', colour='white'), 
+        legend_title=pn.element_text(family='sans-serif', size=15),
+        legend_text=pn.element_text(family='sans-serif', size=12),
+        plot_title=pn.element_text(family='sans-serif', size=15),
+        axis_text=pn.element_text(family='sans-serif', size=12),
+        axis_title=pn.element_text(family='sans-serif', size=15)
+        )
+
+    print(fig)
+    
+    # Save figure
+    fig.save(
+        output_figure_filename,
+        format="svg",
+        bbox_inches="tight",
+        transparent=True,
+        pad_inches=0,
+        dpi=300,
+    )
+
+
+# In[13]:
+
+
+plot_significance_vs_ranking(gsea_pathway_summary, "GSEA", "adjusted p-value (BH)", "GSEA_pathway_ranking.svg")
+
+
+# In[14]:
+
+
+plot_significance_vs_ranking(gsva_pathway_summary, "GSVA", "Enrichment score", "GSVA_pathway_ranking.svg")
+
+
+# In[15]:
+
+
+plot_significance_vs_ranking(roast_pathway_summary, "ROAST", "FDR (BH)", "ROAST_pathway_ranking.svg")
+
+
+# In[16]:
+
+
+plot_significance_vs_ranking(camera_pathway_summary, "CAMERA", "FDR (BH)", "CAMERA_pathway_ranking.svg")
+
+
+# In[17]:
+
+
+plot_significance_vs_ranking(ora_pathway_summary, "ORA", "adjusted p-value (BH)", "ORA_pathway_ranking.svg")
+
+
+# **Takeaway:**
+# * Here are the results demonstrating that different enrichment methods can easily be plugged into our simulation workflow to identify generic gene sets
+# * Overall, it appears that the most enrichment gene sets are also those that are most commonly found to be enriched. However, depending on the enrichment method, generic gene sets will vary slightly due to the different assumptions and modeling procedures. More details about the methods can be found in the [previous notebook](2_apply_enrichment_method.ipynb)
+# * ROAST performs a self-contained test, which assesses the relevance of an individual biological process to the experiment at hand without reference to other genes in the genome. This lack of reference might account for why the scores are uniform across the different gene sets.
```

Comment:
Yeah, it's weird that ROAST doesn't distinguish between gene sets at all! Wouldn't the rankings (y-axis) be different between gene sets, though? Or are the gene sets you're using different for this experiment?

Reply:
This is the best explanation I could come up with about the ROAST results. But all the methods used the same gene sets. My guess is that this method isn't appropriate here for some reason...maybe the number of gene sets is too many. I'll have to think more about this but I believe I've implemented it correctly.